diff -up libgap-4.6.5/test/shell.c.orig libgap-4.6.5/test/shell.c
--- libgap-4.6.5/test/shell.c.orig	2013-10-05 15:38:25.000000000 -0300
+++ libgap-4.6.5/test/shell.c	2013-10-08 13:58:30.693530447 -0300
@@ -20,10 +20,12 @@ extern char **environ;
 
 void print_stack_start()
 {
+#if defined(__i386__) || defined(__x86_64__)
   register void* ebp asm("ebp");     
   int i;
   printf("ebp = %p\n", ebp);
   printf("&i = %p\n", &i);
+#endif
   printf("frame addres = %p\n", __builtin_frame_address(0));
 }
 
